File: services/backend/plan_service/src/main/java/com/example/plan_service/presentation/controllers/HealthController.java
package com.example.plan_service.presentation.controllers;

import com.example.plan_service.presentation.http.Response;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.scheduling.annotation.Async;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.concurrent.CompletableFuture;

@RestController
@RequestMapping("/api/planning")
public class HealthController {
    @Async
    @RequestMapping(value = "/ping", method = RequestMethod.GET)
    @ResponseBody
    public CompletableFuture<ResponseEntity<Object>> ping(){
        Response<Object> result = new Response<>(true, 200, null, "", "");
        return CompletableFuture.completedFuture(new ResponseEntity<>(result, HttpStatus.valueOf(result.getCode())));
    }
}

File: services/backend/user_service/src/main/kotlin/com/example/user_service/presentation/controllers/HealthController.kt
package com.example.user_service.presentation.controllers

import com.example.user_service.presentation.http.Response
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.scheduling.annotation.Async
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/users/ping")
class HealthController {
    @Async
    @RequestMapping("", method=[RequestMethod.GET])
    @ResponseBody
    fun ping(): ResponseEntity<Any> {
        return ResponseEntity
            .status(HttpStatus.OK)
            .body(Response(successfulOperation = true, code = 200, data = null))
    }
}